oak_restricted_kernel_dice/src/lib.rs

```diff
@@ -14,6 +14,15 @@
 // limitations under the License.
 //
 
+//! This crate contains the logic used by oak_restricted_kernel to create attestations. It is broken
+//! out into a seperate crate to allow this logic to be used independently used in tests to create
+//! mock attestations, without also pulling in oak_restricted_kernel's allocator, which breaks
```

Comment:
I am still not sure I understand why this conditional annotation does not work as intended:

https://github.com/project-oak/oak/blob/dbc2bb1c7b83e64dcbfbe79c855f788bc673c0e9/oak_restricted_kernel/src/memory.rs#L38-L40

If it's useless, should we remove it?

Reply:
it

> I am still not sure I understand why this conditional annotation does not work as intended:
> 
> https://github.com/project-oak/oak/blob/dbc2bb1c7b83e64dcbfbe79c855f788bc673c0e9/oak_restricted_kernel/src/memory.rs#L38-L40
> 
> If it's useless, should we remove it?

oh the reason is that this annotation does not work when depending on the crate in other crate's tests. Then that flag is false. We could add feature flag to accomplish the same across crate boundaries, if you prefer that over splitting the crate.

Reply:
Ah I see, thanks for the clarification. The extra feature flag may still break when building the entire workspace at once, since IIRC the features are all merged together before compiling. Anyways maybe we can revisit in the future, @conradgrobler and @andrisaar may have other suggestions too

Reply:
I would assume that hence this annotation would not be useless, it's still needed by tests run within this crate.

Reply:
I did in fact try the option of using a feature flag as an alternative. It works fine, you can see the code here: https://github.com/project-oak/oak/compare/main...jul-sh:feature-flag-alloc-for-use-in-tests?expand=1

The diff is a lot smaller, but I think conceptually it's a bad idea.